### PR TITLE
Refacto/rebuild suggestor

### PR DESCRIPTION
**Pull request by eskombro**
Suggestor has been rebuilt to reuse tree instead of creating a new one for each suggestion.

After a player moves, if the state has already been analyzed in the tree, we just add a new layer to the three and launch minimax. If it isn't, then we create a new tree

### CODE REVIEW

File: internal/brain/suggestor.go [hunk 1]
```diff
@@ -3,123 +3,127 @@ package brain
...
+		}
 	}
 }
 
-func build_tree(depth int16, playerId int16) {
-	//Create tree root
-	tree = tr.Node{Id: 1, Value: 0, Tab: Game.Goban.Tab, Player: Game.CurrentPlayer.Id}
-	tree.Captured[1] = Game.P1.CapturedPieces
-	tree.Captured[2] = Game.P2.CapturedPieces
+func reuseTree(depth int16, playerId int16, lastMove int16) {
```
Comment (ekelen) [inline internal/brain/suggestor.go:66]: Sweet

File: internal/brain/suggestor_utils.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	new.Tab = node.Tab
+	new.Tab[move] = playerId
+	new.Captured[1] = node.Captured[1]
+	new.Captured[2] = node.Captured[2]
+
+	// Virtual Capturing
+	captureDirections := checkCapture(move, &node.Tab, playerId)
+	new.Captured[playerId] += 2 * int16(len(captureDirections))
+	capturePairs(move, captureDirections, &new.Tab)
+	new.Value = getHeuristicValue(playerId, &new.Tab, &new.Captured)
```
Comment (ekelen) [inline internal/brain/suggestor_utils.go:25]: It's okay heuristics, you'll be as great as me someday

File: internal/brain/heuristics.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ const (
 //Add tests
 //Add tests
 //Add tests
```
Comment (ekelen) [inline internal/brain/heuristics.go:24]: //Add tests

**Review by ekelen — APPROVED** [review]
Sweet 👍

File: internal/brain/heuristics.go (hunk 3 above)
Comment (eskombro) [reply internal/brain/heuristics.go:24]: //Add tests

Comment (eskombro) [reply internal/brain/heuristics.go:24]: //Add tests

Comment (eskombro) [reply internal/brain/heuristics.go:24]: //Add tests

**Pull request merged**